src/com/esotericsoftware/yamlbeans/Beans.java

```diff
@@ -38,253 +38,370 @@
+			} else if (Map.class.isAssignableFrom(type)) {
+				commonImplementationConstructor = HashMap.class.getConstructor(noArgs);
 			}
-			return new FieldProperty(field);
+		} catch (Exception e) {
+			throw new InvocationTargetException(e, "Error getting constructor for class: " + type.getName());
 		}
-		return null;
+
+		return commonImplementationConstructor;
 	}
 
-	static private ArrayList<Field> getAllFields (Class type) {
-		ArrayList<Class> classes = new ArrayList();
+	static private ArrayList<Field> getAllFields(Class type) {
+		ArrayList<Class> classes = new ArrayList<Class>();
+		ArrayList<Field> allFields = new ArrayList<Field>();
+
 		Class nextClass = type;
 		while (nextClass != null && nextClass != Object.class) {
 			classes.add(nextClass);
 			nextClass = nextClass.getSuperclass();
 		}
-		ArrayList<Field> allFields = new ArrayList();
+
 		for (int i = classes.size() - 1; i >= 0; i--) {
 			Collections.addAll(allFields, classes.get(i).getDeclaredFields());
 		}
+
 		return allFields;
 	}
 
+	static private String getUpperName(String name) {
+		return Character.toUpperCase(name.charAt(0)) + name.substring(1);
+	}
+
+	static private boolean isBooleanType(Field field) {
+		return field.getType().equals(Boolean.class) || field.getType().equals(boolean.class);
+	}
+
+	static private boolean isProperField(Field field, boolean privateFields) {
+		boolean isProper;
+
+		int modifiers = field.getModifiers();
+		if (Modifier.isStatic(modifiers) || Modifier.isTransient(modifiers)) {
+			isProper = false;
+		} else if (!Modifier.isPublic(modifiers) && !privateFields) {
+			isProper = false;
+		} else {
+			isProper = true;
+		}
+
+		return isProper;
+	}
+
 	static public class MethodProperty extends Property {
-		private final Method setMethod, getMethod;
 
-		public MethodProperty (String name, Method setMethod, Method getMethod) {
+		private final Method setMethod;
+		private final Method getMethod;
+
+		public MethodProperty(String name, Method setMethod, Method getMethod) {
 			super(getMethod.getDeclaringClass(), name, getMethod.getReturnType(), getMethod.getGenericReturnType());
 			this.setMethod = setMethod;
 			this.getMethod = getMethod;
 		}
 
-		public void set (Object object, Object value) throws Exception {
+		public void set(Object object, Object value) throws Exception {
 			if (object instanceof DeferredConstruction) {
-				((DeferredConstruction)object).storeProperty(this, value);
-				return;
+				((DeferredConstruction) object).storeProperty(this, value);
+			} else {
+				setMethod.invoke(object, value);
 			}
-			setMethod.invoke(object, value);
 		}
 
-		public Object get (Object object) throws Exception {
+		public Object get(Object object) throws Exception {
 			return getMethod.invoke(object);
 		}
 	}
 
 	static public class FieldProperty extends Property {
+
 		private final Field field;
 
-		public FieldProperty (Field field) {
+		public FieldProperty(Field field) {

```

Comment:
Formatting for PRs should follow the project's style. This Eclipse formatter can be used:
https://github.com/libgdx/libgdx/blob/master/eclipse-formatter.xml